File: backend/src/main/java/edu/cit/spot/entity/Seat.java
package edu.cit.spot.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Entity
@Data
@Table(name = "seats")
public class Seat {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "section_id", nullable = false)
    private Section section;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;
    
    @NotNull
    @Column(name = "seat_column")
    private Integer column;
    
    @NotNull
    @Column(name = "seat_row")
    private Integer row;
    
    @Version
    private Long version;
}

File: frontend_mobile/app/src/main/java/com/example/spot/model/Seat.kt
package com.example.spot.model

/**
 * Seat data model matching the backend SeatDto
 */
data class Seat(
    val id: Long,
    val sectionId: Long,
    val student: Student?,
    val column: Int,
    val row: Int
) {
    // Computed property to check if seat is taken
    val isTaken: Boolean
        get() = student != null
}
